runners/google-cloud-dataflow-java/build.gradle

```diff
@@ -227,6 +227,34 @@ task validatesRunner {
   dependsOn validatesRunnerLegacyWorkerTest
 }
 
+task validatesRunnerV2Test(type: Test) {
+  group = "Verification"
+  dependsOn buildAndPushDockerContainer
+
+  systemProperty "beamTestPipelineOptions", JsonOutput.toJson([
+          "--runner=TestDataflowRunner",
+          "--project=${dataflowProject}",
+          "--region=${dataflowRegion}",
+          "--tempRoot=${dataflowValidatesTempRoot}",
+          "--workerHarnessContainerImage=${dockerImageContainer}:${dockerTag}",
+          "--experiments=beam_fn_api,use_unified_worker",
+  ])
+
+  // Increase test parallelism up to the number of Gradle workers. By default this is equal
+  // to the number of CPU cores, but can be increased by setting --max-workers=N.
+  maxParallelForks Integer.MAX_VALUE
+  classpath = configurations.validatesRunner
+  testClassesDirs = files(project(":sdks:java:core").sourceSets.test.output.classesDirs) +
+          files(project(project.path).sourceSets.test.output.classesDirs)
+  useJUnit {
+    includeCategories 'org.apache.beam.sdk.testing.ValidatesRunner'
+    excludeCategories 'org.apache.beam.sdk.testing.RunnerV2Incompatible'
+    commonExcludeCategories.each {
```

Comment:
Dataflow Runner V2 supports several items in the commonExcludeCategories like map state, bundle finalizer, test stream. Please only exclude tests that are excluded within google.

Reply:
I will make the stronger statement that I prefer to not use a "commonExcludeCategories" as it is not self documenting and not causal. Only if there is a shared _cause_ for not supporting something should you share the list of exclusions. Otherwise each exclusion overlaps only by coincidence and should not be shared but listed separately.